lib/share_progress/configuration.rb

```diff
@@ -0,0 +1,22 @@
+module ShareProgress
+  def self.configuration
+    @config || Configuration.new
+  end
+
+  def self.configuration=(configuration)
+    @config = configuration
+  end
+
+  def self.configure
+    yield configuration
+  end
+
+  class Configuration
+    attr_accessor :share_progress_uri, :share_progress_api_key
+
+    def initialize
+      @share_progress_uri = ENV['SHARE_PROGRESS_URI'] || 'run.shareprogress.org/api/v1'
+      @share_progress_api_key = ENV['SHARE_PROGRESS_API_KEY'] || 'please_set_an_api_key'
```

Comment:
Addendum: it also makes testing a lot easier. You can't run VCR tests correctly with an invalid key, and it's much easier to handle masked valid API keys using the environment than setting up a whole `Settings` setup which shares git ignored files just for testing. 

Is there a reason to cut the `ENV` default other than just that we don't really do most things that way anymore?